### PR TITLE
Multiplier generics

### PR DESCRIPTION
**Pull request by MartinMystikJonas**
Added generics for Multipler to be able to specify what type created components are

### CODE REVIEW

**Comment by ondrejmirtes** [comment]
1) Please add TypeInferenceTestCase to verify it works as expected.
2) Please add the class to skipCheckGenericClasses, see phpstan-doctrine for an example.

**Comment by MartinMystikJonas** [comment]
Updated s requested

File: stubs/Application/UI/Multiplier.stub [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+namespace Nette\Application\UI;
+
+/**
+ * @template T of \Nette\ComponentModel\IComponent
+ */
+final class Multiplier extends Component
+{
+    /**
+     * @param callable(string, self<T>) : T|callable(string) : T $factory
```
Comment (ondrejmirtes) [inline stubs/Application/UI/Multiplier.stub]: Can we drop the union if we mark the 2nd parameter as optional with `=`?

**Review by ondrejmirtes — CHANGES_REQUESTED** [review]

File: stubs/Application/UI/Multiplier.stub (hunk 1 above)
Comment (MartinMystikJonas) [reply stubs/Application/UI/Multiplier.stub]: Would not it require callable signatire to also always have optional second parameter?

Comment (ondrejmirtes) [reply stubs/Application/UI/Multiplier.stub]: Try it ;) phpstan.org/try

Comment (MartinMystikJonas) [reply stubs/Application/UI/Multiplier.stub]: Ok it works even without =. Updated

**Review by ondrejmirtes — APPROVED** [review]

**Comment by ondrejmirtes** [comment]
Thank you!

**Pull request merged**